pkg/processor/runnable_processor.go

```diff
@@ -59,19 +62,41 @@ func (p *RunnableProcessor) Process(ctx context.Context, records []opencdc.Recor
 		p.inInsp.Send(ctx, record.FromOpenCDC(inRec))
 	}
 
-	outRecs := p.proc.Process(ctx, records)
-	for _, outRec := range outRecs {
-		singleRec, ok := outRec.(sdk.SingleRecord)
+	out := make([]sdk.ProcessedRecord, 0, len(records))
+	for _, rec := range records {
+		keep, err := p.evaluateCondition(rec)
+		if err != nil {
+			return append(out, sdk.ErrorRecord{Error: cerrors.Errorf("failed evaluating condition: %w", err)})
+		}
+		if !keep {
+			out = append(out, sdk.FilterRecord{})
+			continue
+		}
+
+		proc := p.proc.Process(ctx, records)
+		singleRec, ok := proc[0].(sdk.SingleRecord)
 		if ok {
 			p.outInsp.Send(ctx, record.FromOpenCDC(opencdc.Record(singleRec)))
 		}
+		// NB: the processor node that is calling this RunnableProcessor
+		// will check if the number of processed records is equal
+		// to the number of input records.
+		out = append(out, proc...)
 	}
 
-	return outRecs
+	return out
 }
 
 func (p *RunnableProcessor) Teardown(ctx context.Context) error {
 	err := p.proc.Teardown(ctx)
 	p.running = false
 	return err
 }
+
+func (p *RunnableProcessor) evaluateCondition(rec opencdc.Record) (bool, error) {
+	if p.cond == nil {
```

Comment:
WDYT about setting the default condition to `true` when enriching the pipeline config file too, in `pkg/provisioning/config/enrich.go`

Reply:
That's a good idea I believe. The only reason that makes me think we should keep it this way is maybe slightly greater performance, since we won't be executing a template at all (even if it's simple).